Document that query cache is enabled by default

rmloveland opened the pull request:
Fixes #4334.

Summary of changes:

- Update 'Query plan cache' section of the CBO page as follows:

  - Query cache is now enabled by default

  - BUGFIX: Updated example to use `SET CLUSTER SETTING`, was using
    `SET`

  - Removed the following limitations since (if I'm reading PRs
    correctly) they no longer exist:

     - Invalidate cached plans when stats
       change (cockroachdb/cockroach#34057)

     - Query cache works for prepared
       stmts (cockroachdb/cockroach#33575)

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/4335)
<!-- Reviewable:end -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/fa67b1bd4036aa2d44c36e17ac16dfdd3b69c2a7/

rmloveland commented:
Direct link to changed section of CBO page is [here](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/fa67b1bd4036aa2d44c36e17ac16dfdd3b69c2a7/dev/cost-based-optimizer.html#query-plan-cache).

Additional questions:

- Did I interpret the PRs correctly that those 2 limitations were removed?  (around prepared stmts and cache invalidation when stats change)
- Is the "simple memory mgmt" limitation still in place?
- Is there any additional information needed that isn't here?

justinj commented:
LGTM

* Yup,
* yes,
* nothing major I can think of.

rmloveland commented:
Thanks for the review @justinj !

RaduBerinde commented on v2.2/cost-based-optimizer.md:
```diff
@@ -127,20 +127,18 @@ To turn on this feature:
-The query plan cache is still under development and has the following limitations:  
-- The cache is only used for non-prepared statements (i.e., queries that correspond to the ["simple query" pgwire message](https://www.postgresql.org/docs/10/protocol-flow.html#id-1.10.5.7.4)).  
+The query plan cache is still under development and has the following limitation:  
 - The cache has simplistic memory management: it uses a fixed number of "slots" and rejects all plans above a certain size.  
```
This is fixed too

rmloveland replied:
Oh nice!  Will remove that note altogether then.

rmloveland commented:
> LGTM. Another thing to mention is that only SELECT / INSERT / UPDATE / UPSERT / DELETE statements utilize the plan cache.

Thanks Radu - I've deleted the previous information about the simple memory management scheme limitation and added this information.

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0e8f9fa835226d61701a37357bc199194cf82b62/
